src/expr/src/vector_op/cast.rs

```diff
@@ -268,6 +269,7 @@ macro_rules! for_each_cast {
             { date, varchar, general_to_string },
             { timestamp, varchar, general_to_string },
             { timestampz, varchar, timestampz_to_utc_string },
+            { list, varchar, |x| general_to_string(x) },
```

Comment:
What's the difference between `general_to_string` and `|x| general_to_string(x)`?

Reply:
It is like the relationship between `min` and `min_str` in batch agg:
https://github.com/risingwavelabs/risingwave/blob/c343684d35347f5cdeea1a3e9f2c4acaecfa6561/src/expr/src/vector_op/agg/functions.rs#L85-L87

When the compiler fills the generic function with a concrete type, a fixed lifetime is chosen at the same time. However, we need a function that works for any lifetime (`for<'a>`).